lib/galaxy/datatypes/registry.py

```diff
@@ -544,6 +557,28 @@ def load_datatype_sniffers(self, root, deactivate=False, handling_proprietary_da
                                         if sniffer_class not in sniffer_elem_classes:
                                             self.sniffer_elems.append(elem)
 
+    def get_datatype_from_filename(self, name):
+        generic_datatype_instance = self.get_datatype_by_extension("data")
+        if "." not in name:
+            return generic_datatype_instance
+        name_parts = name.rsplit(".", 3)
+        possible_extensions = []
+        for extension_segments in [3, 2, 1]:
+            if len(name_parts) < extension_segments + 1:
+                continue
+            possible_extensions.append(".".join(name_parts[-extension_segments:]))
+
+        infer_from = self.datatypes_by_suffix_inferences
+        for possible_extension in possible_extensions:
+            if possible_extension in infer_from:
+                return infer_from[possible_extension]
+
+        for possible_extension in possible_extensions:
+            if possible_extension in self.datatypes_by_extension:
+                return self.datatypes_by_extension[possible_extension]
```

Comment:
Is this the intended order of lookups? Giving `infer_from` priority over `datatypes_by_extension`?

Reply:
In the case of inferring things from the filename - yes - I did intend to do that. The Galaxy "extension" doesn't really map well to actual file extensions even for really basic things like fastq and fasta files.